hardware/mifare/mifare_classic.ksy

```diff
@@ -0,0 +1,244 @@
+meta:
+  id: mifare_classic
+  endian: le
+  file-extension: mfd
+  title: "Mifare Classic RFID tag dump"
+  license: BSD-2-Clause
+doc-ref: |
+  https://github.com/nfc-tools/libnfc
+  https://www.nxp.com/docs/en/data-sheet/MF1S70YYX_V1.pdf
+doc: |
+  You can get a dump for testing by the link: https://github.com/zhovner/mfdread/raw/master/dump.mfd
+seq:
+  - id: sectors
+    size: ((_index >= 32)?4:1)*4*16 #sorry for this doubling of `block_size` (16), but we need `sector` be self-sufficient so we cannot use _root there
+    type: sector(_index == 0)
+    repeat: eos
+types:
+  key:
+    seq:
+      - id: key
+        size: 6
+  sector:
+    params:
+      - id: has_manufacturer
+        type: bool
+    seq:
+      - id: manufacturer
+        type: manufacturer
+        if: has_manufacturer
+      
+      - id: data_filler
+        -orig-id: abtData
+        size: _io.size - _io.pos - 16 # sizeof(trailer)
+        type: filler
+      - id: trailer
+        type: trailer
+    instances:
+      block_size:
+        value: 16
```

Comment:
If this is a global constant, it makes sense to move it to top level type (i.e. `mifare_classic`)?

Reply:
>If this is a global constant, it makes sense to move it to top level type (i.e. mifare_classic)?

It is completely makes sense to move (in fact initially it was there), but if I do it `sector` will depend on presence of it in `_root` and there will be troubles in using `sector` separately.

It may make sense to a add compiler optimization detecting constants and moving them into global scope and using correzponding dzta types for them.

Reply:
> It may make sense to a add compiler optimization detecting constants and moving them into global scope and using corresponding data types for them.

Actually, this is a valid reason to finally introduce something like `consts` (on the same level with `seq`, `instances`, etc).